app/src/main/java/org/jellyfin/mobile/media/car/LibraryBrowser.kt

```diff
@@ -245,143 +235,162 @@ class LibraryBrowser(
-            id = playlistId
-        }
-        return apiClient.getPlaylistItems(query)?.extractItems("${LibraryPage.PLAYLIST}|$playlistId")
+    private suspend fun getPlaylist(playlistId: UUID): List<MediaMetadataCompat>? {
+        val result by playlistsApi.getPlaylistItems(
+            playlistId = playlistId,
+            userId = appPreferences.currentUserUuid ?: return null
+        )
+
+        return result.extractItems("${LibraryPage.PLAYLIST}|$playlistId")
     }
 
-    private fun ItemsResult.extractItems(libraryId: String? = null): List<MediaMetadataCompat> =
-        items.map { item -> buildMediaMetadata(item, libraryId) }.toList()
+    private fun BaseItemDtoQueryResult.extractItems(libraryId: String? = null): List<MediaMetadataCompat>? =
+        items?.map { item -> buildMediaMetadata(item, libraryId) }?.toList()
 
     private fun buildMediaMetadata(item: BaseItemDto, libraryId: String?): MediaMetadataCompat {
         val builder = MediaMetadataCompat.Builder()
         builder.setMediaId(buildMediaId(item, libraryId))
         builder.setTitle(item.name ?: context.getString(R.string.media_service_car_item_no_title))
 
         val isAlbum = item.albumId != null
-        val imageOptions = ImageOptions().apply {
-            imageType = ImageType.Primary
-            maxWidth = 1080
-            quality = 90
-            tag = if (isAlbum) item.albumPrimaryImageTag else item.imageTags[ImageType.Primary]
-        }
-        val primaryImageUrl = when {
-            item.hasPrimaryImage -> apiClient.GetImageUrl(item, imageOptions)
-            isAlbum -> apiClient.GetImageUrl(item.albumId, imageOptions)
+        val itemId = when {
+            item.imageTags.containsKey(ImageType.PRIMARY) -> item.id
+            isAlbum -> item.albumId
             else -> null
         }
+        val primaryImageUrl = itemId?.let {
+            imageApi.getItemImageUrl(
+                itemId = itemId,
+                imageType = ImageType.PRIMARY,
+                tag = if (isAlbum) item.albumPrimaryImageTag else item.imageTags[ImageType.PRIMARY]
+            )
+        }
+
+        if (item.type.equals("audio", ignoreCase = true)) {
+            val uri = universalAudioApi.getUniversalAudioStreamUrl(
+                itemId = item.id,
+                userId = appPreferences.currentUserUuid,
+                deviceId = deviceInfo.id,
+                maxStreamingBitrate = 140000000,
+                container = listOf(
+                    "opus",
+                    "mp3|mp3",
+                    "aac",
+                    "m4a",
+                    "m4b|aac",
+                    "flac",
+                    "webma",
+                    "webm",
+                    "wav",
+                    "ogg"
+                ),
+                transcodingContainer = "ts",
+                transcodingProtocol = "hls",
+                audioCodec = "aac",
+                // TODO missing: PlaySessionId=randomUUID()
```

Comment:
These 1 properties don't exist in the current API, they probably didn't in the old one either so they can probably be removed without any issue.

Reply:
We should remove those - they probably didn't do anything before.